packages/nuxi/src/commands/dev.ts

```diff
@@ -44,20 +44,27 @@ export default defineNuxtCommand({
 
     await setupDotenv({ cwd: rootDir, fileName: args.dotenv })
 
+    const { loadNuxt, loadNuxtConfig, buildNuxt } = await loadKit(rootDir)
+
+    const config = await loadNuxtConfig({
+      cwd: rootDir,
+      overrides: { dev: true }
+    })
+
     const listener = await listen(serverHandler, {
       showURL: false,
       clipboard: args.clipboard,
       open: args.open || args.o,
-      port: args.port || args.p || process.env.NUXT_PORT,
-      hostname: args.host || args.h || process.env.NUXT_HOST,
-      https: args.https && {
-        cert: args['ssl-cert'],
-        key: args['ssl-key']
-      }
+      port: args.port || args.p || process.env.NUXT_PORT || config.devServer.port,
+      hostname: args.host || args.h || process.env.NUXT_HOST || config.devServer.host,
+      https: (args.https || config.devServer.https)
```

Comment:
Here we should check `args.https !== false` as well. Disabling https if explicit `--no-https` flag was passed to CLI otherwise it is impossible to disable https without updating config.